Services/Utilities/classes/class.ilUtil.php

```diff
@@ -189,23 +189,29 @@ public static function getStyleSheetLocation($mode = "output", $a_css_name = "",
 		}
 
 		$filename = "";
+		$ver_filename = "";
 		// use ilStyleDefinition instead of account to get the current skin
 		require_once("./Services/Style/classes/class.ilStyleDefinition.php");
 		if (ilStyleDefinition::getCurrentSkin() != "default")
 		{
 			$filename = "./Customizing/global/skin/".ilStyleDefinition::getCurrentSkin()."/".$a_css_location.$stylesheet_name;
+			$ver_filename = "./Customizing/global/skin/".ilStyleDefinition::getCurrentSkin()."/version";
```

Comment:
I am unsure wheter a "version" file is the way to go. This is very implicit. Is this a common pattern in other applications? Why not adding a seperate field to a styles xml? This would have the nice side effect that we have the whole style under "version-control", this could then also be used for the icons (which suffer a similar problem as the css does at the current state).

Reply:
So with 5.2 the editor would automatically increase the version in styles.xml? Or should this even be exposed to that user allowing to enter this values via UI?

Reply:
Thank you two very much, for your suggestions.

I created a second PR for ILIAS 5.2 with another approach and support for the UI skin editor. The full informations are in the PR description: https://github.com/ILIAS-eLearning/ILIAS/pull/778

To answer your questions:
I originally used the version file, because I wanted to get the number of code changes as low as possible. You are right that this is not an optimal way but I was afraid, that the bugfix could be declined because it could be seen as an feature instead of an bugfix, if the code changed to much. Also in this fix I did not think about the UI skin editor.

Now, after I made the bugfix for ILIAS 5.2, I will take a look how to integrate "the better way" (use of the skins template.xml) into the other versions. Maybe I will do a second PR after on ILIAS 5.0 with the alternative bugfix. Or would you prefer that I just update this one?

Also I got a question about the documentations mentioned by @Amstutz before. This files seems to only exist in the branches release_5-3 and trunk. Should I really edit or add this files, because of the bugfix for version 5.0, if they are not part of this version?
I would suggest to wait after the JF and make an PR for the documentation, based on the bugfix version that will be accepted. What do you think?